axlearn/common/trainer.py

```diff
@@ -1026,6 +1042,9 @@ def _get_compiled_train_step_fn(
             )
             return self._compiled_train_step
         logging.log_first_n(logging.INFO, "Compiling XSC train step.", 1)
+        self._maybe_record_event(
+            measurement.Event.START_CUSTOM_BADPUT_EVENT, custom_badput_event_type="SDC_CHECK"
+        )
```

Comment:
We should check if the [xsc_policy_check](https://github.com/apple/axlearn/blob/121f43206facb2c9a9a524dbf5774c7d07bfdffb/axlearn/common/trainer.py#L170) is enabled before recording this. Perhaps inside the compile function would be best.

Ethan confirmed that it is not enabled by default.

Thanks for the pointers @Ethanlm!

Reply:
@dipannita08 Would it be easier to revert the recent change, get this PR merged first, and make a separate PR for SDC related change?

Reply:
It would be easier (overall faster) to get the bigger change and the SDC + Eval Step support merged in together.

But if we are looking to restart the job over the weekend, then I'm good to revert the latest changes, merge this PR and create a new one for the SDC + Eval changes!

Reply:
That's fine. We are not going to add this to the job. It is going to finish pretty soon. I asked mainly for easier reviews. Thanks

Reply:
Sounds good!